packages/bundler-plugin-core/src/utils/normalizeOptions.ts

```diff
@@ -0,0 +1,72 @@
+import { type Options } from "../types.ts";
+import { z } from "zod";
+
+export type NormalizedOptions = z.infer<typeof optionsSchema> & Options;
+
+const validBundleName = /^[\w\d_:/@\.{}\[\]$-]+$/;
+
+const optionsSchema = z.object({
+  apiUrl: z.string().url().default("https://api.codecov.io"),
+  dryRun: z.boolean().default(false),
+  bundleName: z.string().regex(validBundleName),
+});
+
+interface NormalizedOptionsFailure {
+  success: false;
+  errors: string[];
+}
+
+interface NormalizedOptionsSuccess {
+  success: true;
+  options: NormalizedOptions;
+}
+
+export type NormalizedOptionsResult =
+  | NormalizedOptionsFailure
+  | NormalizedOptionsSuccess;
+
+export const normalizeOptions = (
+  userOptions: Options,
+): NormalizedOptionsResult => {
+  const validatedOptions = optionsSchema.safeParse(userOptions);
+
+  if (!validatedOptions.success) {
+    const errorMessages: string[] = [];
+    const issues = validatedOptions.error.issues;
+
+    for (const issue of issues) {
+      if (issue.path.at(0)?.toString() === "bundleName") {
```

Comment:
Is it possible to add a https://zod.dev/ERROR_HANDLING?id=schema-bound-error-map? Then we can just directly return messages?